File: src/Entity/Dossier.php
<?php

namespace App\Entity;

use App\Repository\DossierRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: DossierRepository::class)]
class Dossier
{
    #[ORM\Id]
    #[ORM\ManyToOne(inversedBy: 'dossiers')]
    #[ORM\JoinColumn(nullable: false, name: "idUser",referencedColumnName:"IdUser")]
    private ?User $idUser = null;

    #[ORM\Id]
    #[ORM\ManyToOne(inversedBy: 'dossiers')]
    #[ORM\JoinColumn(nullable: false, name: "idOffre",referencedColumnName:"idOffre")]
    private ?Offre $idOffre = null;

    #[ORM\Column(name:"status",length: 20)]
    private ?string $status = null;

    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): static
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function getIdOffre(): ?Offre
    {
        return $this->idOffre;
    }

    public function setIdOffre(?Offre $idOffre): static
    {
        $this->idOffre = $idOffre;

        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;

        return $this;
    }
}

File: src/main/java/com/example/javafxprojectusertask/Entities/Dossier.java
package com.example.javafxprojectusertask.Entities;

public class Dossier {


    String status;

    Offre offre;

    int idUser;


    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Offre getOffre() {
        return offre;
    }

    public void setOffre(Offre offre) {
        this.offre = offre;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    @Override
    public boolean equals(Object o)
    {
        if(o == null)
            return false;
        if(o instanceof Dossier d)
        {
            return this.idUser == d.getIdUser();
        }
        return false;
    }


    public String getClassName()
    {
        return "Dossier";
    }
}
